docs/proposals/20191017-kubeadm-based-control-plane.md

```diff
@@ -472,13 +472,21 @@ When `MaxSurge` is set to 0 the rollout algorithm is as follows:
   for additional details. When there are multiple machines that are marked for remediation, the oldest one will be remediated first.
 
 - Following rules should be satisfied in order to start remediation
-  - The cluster MUST have at least two control plane machines, because this is the smallest cluster size that can be remediated.
-  - The number of replicas MUST be equal to or greater than the desired replicas. This rule ensures that when the cluster
-    is missing replicas, we skip remediation and instead perform regular scale up/rollout operations first.
+  - One of the following apply:
+    - The cluster MUST one machine not yet initialized (the failure happens before KCP reaches the initialized state) 
+    - The cluster MUST have at least two control plane machines, because this is the smallest cluster size that can be remediated.
+  - Previous remediation (delete and re-create) MUST have been completed. This rule prevents KCP to remediate more machine while the
+    replacement for the previous machine is not yet created. 
   - The cluster MUST have no machines with a deletion timestamp. This rule prevents KCP taking actions while the cluster is in a transitional state.
   - Remediation MUST preserve etcd quorum. This rule ensures that we will not remove a member that would result in etcd
     losing a majority of members and thus become unable to field new requests.
 
+- Additionally following opt-in safeguard will be put in place:
+  - If we are remediating the same machine (delete, re-create, replacement machine gets unhealthy), it will be possible
+    to define a maximum number of retries, thus preventing un-necessary load on infrastructure provider e.g. in case of quota problems.
+  - If we are remediating the same machine (delete, re-create, replacement machine gets unhealthy), it will be possible
+    to define a delay between each retry, thus allowing the infrastructure provider to stabilize in case of temporary problems.
```

Comment:
```suggestion
  - If we are remediating the same machine (delete, re-create, replacement machine gets unhealthy), it will be possible
    to define a maximum number of retries, thus preventing unnecessary load on infrastructure provider e.g. in case of quota problems.
  - If we are remediating the same machine (delete, re-create, replacement machine gets unhealthy), it will be possible
    to define a delay between each retry, thus allowing the infrastructure provider to stabilize in case of temporary problems.
```